inst/shiny-server/server/ERT_summary.R

```diff
@@ -45,42 +48,53 @@ runtime_summary <- reactive({
   fstop <- format_FV(input$RTSummary.Statistics.Max) %>% as.numeric
   fstep <- format_FV(input$RTSummary.Statistics.Step) %>% as.numeric
   data <- DATA()
-
-  req(fstart <= fstop, fstep <= fstop - fstart, data)
-  fall <- get_funvals(data)
-
-  if (input$RTSummary.Statistics.Single)
-    fstop <- fstart
-
-  fseq <- seq_FV(fall, fstart, fstop, fstep)
-  req(fseq)
-
+  
+  if (!input$RTSummary.Statistics.Single){
+    req(fstart <= fstop, fstep <= fstop - fstart, length(data) > 0)
+    fall <- get_funvals(data)
+    fseq <- seq_FV(fall, fstart, fstop, fstep)
+    req(fseq)
+  }
+  else{
+    fseq <- fstart
+  }
   df <- get_RT_summary(data, fseq, algorithm = input$RTSummary.Statistics.Algid)
   df <- df[, c('DIM', 'funcId') := NULL]
-  df$target <- format_FV(df$target)
+  df$target <- format_FV(df$target) %>% as.numeric
 
   # format the integers
   probs <- getOption("IOHanalyzer.quantiles")
   for (p in paste0(probs * 100, '%')) {
     df[[p]] %<>% as.integer
   }
+  df$mean <- round(df$mean, digits = 2)
+  df$sd <- round(df$sd, digits = 2)
+  df$ERT <- round(df$ERT, digits = 2)
+  df$ps <- round(df$ps, digits = 2)
+  if (format == COCO)
+    df$target <- formatC(df$target, format = "e", digits = 2)
+
+
   df
 
 })
 
-output$table_RT_summary <- renderDataTable({
+output$table_RT_summary <- DT::renderDataTable({
```

Comment:
What is the difference brought by DT::renderDataTable?

Reply:
The DT-package has its own implementation of the rendering of datatables, which allows for more customization and a way better search than the standard shiny version. To make sure that the correct version of the function is used, the DT:: prefix is used here.